### PR TITLE
Openapi docs

### PR DESCRIPTION
**Pull request by pettinarip**
Defines the API docs following the openapi standard and adds a script to generate them.

Next steps: we can autogenerate a ui using the generated json file using something like https://www.npmjs.com/package/swagger-ui-react

### CODE REVIEW

File: lib/openapi/index.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import { createDocument } from "zod-openapi"
+
+import { machinesPaths } from "./machines"
+import { proofsPaths } from "./proofs"
+
+export const document = createDocument({
+  openapi: "3.1.0",
+  info: {
+    title: "Ethproofs API",
```
Comment (wackerow) [inline lib/openapi/index.ts:9]: @pettinarip Thoughts on using SITE_NAME from lib/constants wherever we mention the Ethproofs name, to keep it consistent with one canonical reference?

Comment (wackerow) [reply lib/openapi/index.ts:9]: I have a separat branch where I was going to update that elsewhere, but was awaiting name confirmation before posting

Comment (wackerow) [reply lib/openapi/index.ts:9]: - #116

Comment (wackerow) [reply lib/openapi/index.ts:9]:
```suggestion
    title: `${SITE_NAME} API`,
```

With import statement: 

```tsx
import { SITE_NAME } from "@/lib/constants"
```

File: lib/openapi/index.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    description: `This document outlines the available API endpoints for ethproofs.\n\n**Authentication**\n\nAll endpoints require authentication using an API key in the request header:\n\n\`Authorization: Bearer <api_key>\``,
+    version: "0.0.1",
+  },
+  servers: [
+    {
+      url: "https://main--ethproofs.netlify.app/api/v0",
```
Comment (wackerow) [inline lib/openapi/index.ts:15]: Similar to above, we have an env var / constant we could potentially use here (`SITE_URL`) to keep things canonical

```suggestion
      url: new URL("api/v0", SITE_URL).toString(),
```

```tsx
import { SITE_URL } from "@/lib/constants"
```

File: public/openapi.json [hunk 3]
```diff
@@ -0,0 +1,333 @@
+{
+  "openapi": "3.1.0",
+  "info": {
+    "title": "Ethproofs API",
+    "description": "This document outlines the available API endpoints for ethproofs.\n\n**Authentication**\n\nAll endpoints require authentication using an API key in the request header:\n\n`Authorization: Bearer <api_key>`",
```
Comment (wackerow) [inline public/openapi.json:5]: Same note as above re: `SITE_NAME`

File: lib/openapi/index.ts (hunk 2 above)
Comment (pettinarip) [reply lib/openapi/index.ts]: Good point. I've added 2 servers for now (prod and preview urls).

**Review by wackerow — APPROVED** [review]

**Pull request merged**